src/lib/controllers/detect-text-controller.js

```diff
@@ -0,0 +1,55 @@
+import { ClientAdapters } from 'alpheios-client-adapters'
+import NotificationSingleton from '@/lib/notifications/notification-singleton'
+import Langs from '@/lib/data/langs/langs.js'
+
+export default class DetectTextController {
+  /**
+   * Checks if text is plain text or xml.
+   * If it is a plain text - than application will send a request to language detection API and defines direction from language.
+   * @param {SourceText} sourceText
+   * @returns {Object}
+   *          {String} sourceType - text/tei
+   *          {String} lang - only for text
+   *          {String} direction - only for text
+   */
+  static async detectTextProperties (sourceText) {
+    const sourceType = this.checkXML(sourceText)

```

Comment:
No, the sense of this method is another.
We support both - plain text and TEI XML.
On this stage - when a user enters a text - we don't check the ability of this text to be tokenized. (we would get it as a result from the remote tokenizer service).

This check is only for the language detection.
We could detect a language only for plain text (if I send an xml - it would be treated as a plain text).
That's why I use this check - to avoid TEI xml for detection.

After reading your comment I think - I would simplify the check and would simply check for `</tei>` tag using `indexOf` method.